samples/basic/hash_map/README.rst

```diff
@@ -16,16 +16,16 @@ This is a simple example that repeatedly:
 Building
 ********
 
-This application can be built on native_posix as follows:
+This application can be built on native_sim as follows:
```

Comment:
Please ignore if you feel likes it's too much of a cosmetic change and too cumbersome in the context of these already tedious PRs, but would be great to have consistency in making references to native_sim a link, just like other samples touches in this PR do

```suggestion
This application can be built on :ref:`native_sim <native_sim>` as follows:
```